pandas-stubs/core/series.pyi

```diff
@@ -1025,12 +1032,96 @@ class Series(IndexOpsMixin, NDFrame, Generic[S1]):
         axis: SeriesAxisType | None = ...,
         ignore_index: _bool = ...,
     ) -> Series[S1]: ...
+    @overload
     def astype(
         self,
-        dtype: S1 | _str | type[Scalar],
+        dtype: S1
+        | pd.Int8Dtype
+        | pd.Int16Dtype
+        | pd.Int32Dtype
+        | pd.Int64Dtype
+        | np.int8
+        | np.int16
+        | np.int32
+        | np.int64
+        | np.uint8
+        | np.uint16
+        | np.uint32
+        | np.uint64
+        | np.intp
+        | np.uintp
+        | np.integer
+        | int
```

Comment:
For all of the `np` types, you have to use `type[np.int8]`, `type[np.int16]` and also use `type[int]` as well.